File: spark/left-semi-join/relevant_users/relevant_users.py
pupil.join(innovationCraft.where(col("type") == 1),
           pupil["id"] == innovationCraft["owner_id"],
           "left_semi") \
     .select("id", "first_name", "last_name", "grade") \
     .orderBy("grade", "id") \
     .show()

File: spark/left-semi-join/relevant_users/relevant_users.scala
pupil.join(innovationCraft.where(col("type") === 1),
           pupil.col("id") === innovationCraft.col("owner_id"),
           "left_semi")
     .select("id", "first_name", "last_name", "grade")
     .orderBy("grade", "id")
     .show()